docs/dev/guidelines/client-tests.rst

```diff
@@ -0,0 +1,253 @@
+      and this logic is important for the component, do not mock the service methods, but mock the HTTP requests and responses from the API. This allows us to test the interaction
+      of the component with the service and in addition test that the service logic works correctly. A good explanation can be found in the `official angular documentation <https://angular.io/guide/http#testing-http-requests>`_.
+
+    .. code:: ts
+
+        import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+        describe('SomeComponent', () => {
+          beforeEach(() => {
+              TestBed.configureTestingModule({
+                  imports: [HttpClientTestingModule],
+              });
+
+              ...
+              httpMock = injector.get(HttpTestingController);
+          });
+
+          afterEach(() => {
+              ...
+              httpMock.verify();
+              sinon.restore();
+          });
+
+          it('should make get request', fakeAsync( () => {
+              const returnedFromApi = {some: 'data'};
+
+              component.callServiceMethod()
+                  .subscribe((data) => expect(data.body).toEqual(returnedFromApi));
+
+              const req = httpMock.expectOne({ method: 'GET', url: 'urlThatMethodCalls' });
+              req.flush(returnedFromApi);
+              tick();
+          }));
+        });
+
+2. Do not overuse ``NO_ERRORS_SCHEMA`` (`link <https://angular.io/guide/testing-components-scenarios#no_errors_schema>`_).
+   This tells angular to ignore the attributes and unrecognized elements, prefer to use component stubs as mentioned above.
+
+3. Sinon uses sandboxes, which remove the need of keeping track of every fake created, which greatly simplifies cleanup and improves readability.
+   Since ``sinon@5.0.0``, the sinon object is a default `sandbox <https://sinonjs.org/releases/latest/sandbox/>`_, meaning one doesn't need to do any setup work.
+   Unless you have a very advanced setup or need a special configuration, you probably want to only use that one.
+   In ``afterEach`` block one should add ``sinon.restore()``, which restores all fakes created through sandbox.
+
+4. Make sure to have at least 80% line test coverage. Running ``yarn test --coverage`` to create a coverage report. You can also simply run the tests in IntelliJ IDEA with coverage activated.
+
+5. It is preferable to test a component through the interaction of the user with the template. This decouples the test from the concrete implementation used in the component.
+   For example if you have a component that loads and displays some data when the user clicks a button, you should query for that button, simulate a click and then assert that the data has been loaded and that the expected
+   template changes have occurred.
```

Comment:
An example for this would be good, but we can do that in a follow-up if you don't want to create one in this PR.

Reply:
I added an example from here: https://github.com/ls1intum/Artemis/blob/master/src/test/javascript/spec/component/shared/exercise-update-warning.component.spec.ts#L38-L49

Do you think it is a good enough example?

Reply:
That looks good to me, thanks. 👍